### ISSUE TITLE
Markdown in room topics should be parsed and displayed properly

### ISSUE THREAD

**Issue by kittykat** [vector-im/element-x-android#613]
### Steps to reproduce

1. Go to a room which has markdown used in a topic (e.g. `[foo](example.com)`)
2. Open room detail view from the top bar

#### What did you expect?
Topic is displayed correctly, like on web

#### What happened instead?
I see the actual markdown which hasn't been parsed

### Will you send logs?

Yes

### Are you willing to provide a PR?

Yes

**Comment by kittykat** [comment]
@VolkerJunginger What are your thoughts on this issue? Should it be a blocker for closing the room detail page story?

**Comment by nadonomy** [comment]
Suggestion - even if we don't implement markdown parsing yet - could be worth converting to plaintext so we don't expose ugly markdown syntax.

**Comment by jmartinesp** [comment]
> Suggestion - even if we don't implement markdown parsing yet - could be worth converting to plaintext so we don't expose ugly markdown syntax.

Actually, that's probably even harder to achieve 😅 . We'd need to use a markdown parser and then remove anything that's not plain text, which is no more work than just using a markdown parser and transforming it to HTML.

**Comment by bmarty** [comment]
Can the conversion of the room topic from markdown to Html be handled by the SDK, either by providing the Html version directly, or by exposing a function to do the conversion? CC @Hywan

**Comment by Hywan** [comment]
That's doable. `ruma` is already doing it for the case of events (`m.text`). We should find a way to expose the Markdown parser used by `ruma`, but clearly doable.

**Comment by jplatte** [comment]
I thought it was mentioned here before, but doesn't seem so: The only way to put HTML into the topic is [MSC3765](https://github.com/matrix-org/matrix-spec-proposals/pull/3765), so we need to add support for that to Ruma first. Just an an implementation hint, @Hywan :)

**Comment by jplatte** [comment]
I don't think the two are related, really (room name and room topic)?

**Comment by Hywan** [comment]
Yeah, I juts got confused. Taking a look now at adding MSC3765 inside Ruma.

**Comment by Hywan** [comment]
The MSC isn't stable, and the implementation inside Element Web doesn't seem to match the MSC. I don't think any of this is ready to be implemented. I think it needs more preparations.

**Comment by bmarty** [comment]
Let try to us the Rich Text Editor library to render the topic.
